tests/test_EclipseIO.cpp

```diff
@@ -431,3 +435,309 @@ WELSPECS
+             edit_mult_x.assign(18, .1);
+         }
+
+         if (doxyz[2].test(1))
+         {
+             deckString += std::string{ R"(
+MULTZ-
+ 18*30 /
+MULTZ-
+ 18*0.3 /
+)"};
+             edit_mult_z.assign(18, .3);
+         }
+         deckString += std::string{ R"(
+EQUALS
+)"};
+         if (doxyz[0].test(0))
+         {
+             deckString += std::string{ R"(
+ 'MULTX' 1.5 3 3 2 2 1 1 / -- Should not overwrite the value of GRID section but of EDIT section
+)"};
+             edit_mult_x[5] = 1.5;
+         }
+
+         if (doxyz[2].test(1))
+         {
+             deckString += std::string{ R"(
+ 'MULTZ-' 3.0 1 1  2 2 1 1 /
+)"};
+             edit_mult_z[3] = 3.0;
+         }
+
+             deckString += std::string{ R"(/
+)"};
+         auto mult = edit_mult_x.begin();
+         if (doxyz[0].test(0)) {
+             for (auto&& val: exspected_multipliers[0][0]) {
+                 val *= *(mult++);
+             }
+         }
+         mult = edit_mult_z.begin();
+         if (doxyz[2].test(1)) {
+             for (auto&& val: exspected_multipliers[2][1]) {
+                 val *= *(mult++);
+             }
+         }
+     }
+
+     return {deckString, exspected_multipliers};
+}
+
+/// \brief Test the MULTXYZ writing
+///
+/// \param doxyz Inidicates for each direction which multipliers should be test
+///              no bit set means none, 1st bit set positive, 2nd bit set negative
+/// \param write_all_mult_minus If true we will request that even defaulted MULT?- arrays will be
+///                  written via first record of GRIDOPTS
+void testMultxyz(std::array<std::bitset<2>,3> doxyz, bool write_all_multminus = false)
+{
+    const auto [deckString, exspectedMult] = createMULTXYZDECK(doxyz, write_all_multminus);
+    const auto deck = Parser().parseString(deckString);
+    auto es = EclipseState( deck );
+    const auto& eclGrid = es.getInputGrid();
+    const Schedule schedule(deck, es, std::make_shared<Python>());
+    const SummaryConfig summary_config( deck, schedule, es.fieldProps(), es.aquifer());
+    const SummaryState st(TimeService::now());
+    es.getIOConfig().setBaseName( "MULTXFOO" );
+    EclipseIO eclWriter( es, eclGrid , schedule, summary_config);
+    eclWriter.writeInitial( );
+
+    EclIO::EclFile initFile { "MULTXFOO.INIT" };
+
+    std::array<std::string, 6> multipliers{"MULTX", "MULTX-", "MULTY", "MULTY-", "MULTZ", "MULTZ-"};
+    int i=0;
+    auto mult = multipliers[i];
+    for (const auto& mult: multipliers) {
+        if (i%2==0 || write_all_multminus || doxyz[i/2].test(1)) {
+            BOOST_CHECK_MESSAGE( initFile.hasKey(mult), R"(INIT file must have ")" + mult + R"(" array)" );
+            const auto& multValues   = initFile.get<float>(mult);
+            auto exspect = exspectedMult[i/2][i%2].begin();
+            BOOST_CHECK(multValues.size() == exspectedMult[i/2][i%2].size());
+            for (auto mult = multValues.begin(); mult != multValues.end(); ++mult, ++exspect)
```

Comment:
This `mult` name shadows the name in the outer scope.  Please choose another name for one of the objects.

Reply:
Fixed

Reply:
The `mult` loop variable still shadows the `mult` loop variable in [the outer loop](https://github.com/OPM/opm-common/blob/5a56bf7a119b6a69c13ca3918eee2583eaaa736f/tests/test_EclipseIO.cpp#L711) and that's preventing us from merging this.